### PR TITLE
Add 'jenkins' module

### PR DESCRIPTION
**Pull request by rhysd**
Please fill in this template.

- [x] Make your PR against the `master` branch.
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped#common-mistakes).
- [x] Run `tsc` without errors.
- [x] Run `npm run lint package-name` if a `tslint.json` is present.

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not provide its own types, and you can not add them.
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Create it with `npm run new-package package-name`, not by basing it on an existing project.
- [x] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, and `strictNullChecks` set to `true`.

If changing an existing definition:
- [ ] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [ ] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "../tslint.json" }`.

If removing a declaration:
- [ ] If a package was never on DefinitelyTyped, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

### CODE REVIEW

**Comment by rhysd** [comment]
There are many 'Avoid using the Function type. Prefer a specific function type, like `() => void`' errors from linter. But it can't be fixable because I'm using overloads in some methods.

```typescript
{
  foo(i: number): void;
  foo(i: number, s: string): void;
}
```

is valid. But,

```typescript
{
  foo: ((i: number): void)
       | ((i: number, s: string): void);
}
```

is not valid on calling the `foo` (`tsc` raises 'cannot call value which is not a function' error).

So I ignored them.

**Comment by rhysd** [comment]
I was told that below seems to resolve this:

```typescript
{
  foo: ((i: number): void)
     & ((i: number, s: string): void);
}
```

However, I'm not feeling this is better than overload syntax (below).

```typescript
{
  foo(i: number): void;
  foo(i: number, s: string): void;
}
```

Should I still fix them with intersection type? Or should I add line comments to disable tslint errors temporarily?

**Comment by andy-ms** [comment]
The linter error says to avoid using the `Function` type. (Not "function types", those are OK. But the capital-F `Function` type should be replaced by particular function types.)  So look for instances of `Function` in your code. It's not telling you to not use overloads.

Although, another lint rule will warn about unnecessary overloads. In the example above, you can use `foo(i: number, s?: string): void;` instead.

**Comment by rhysd** [comment]
ah, I see. Thank you for correcting me. I'll try it later (but actually there is no documentation about the actual type of the `Function`s. So it maybe take a time...)

**Comment by rhysd** [comment]
@andy-ms 

I fixed all issues reported by linter. Could you review this?

File: jenkins/tsconfig.json [hunk 1]
```diff
@@ -0,0 +1,20 @@
+{
+    "compilerOptions": {
+        "module": "commonjs",
+        "target": "es6",
+        "noImplicitAny": true,
+        "noImplicitThis": true,
+        "strictNullChecks": false,
```
Comment (andy-ms) [inline jenkins/tsconfig.json]: Make this `true`

**Comment by rhysd** [comment]
@andy-ms I fixed the point.

**Pull request merged**